src/codegen/child_properties.rs

```diff
@@ -32,6 +33,9 @@ fn generate_func(w: &mut Write, env: &Env, prop: &ChildProperty, in_trait: bool,
 
     try!(writeln!(w, ""));
 
+    if prop.doc_hidden {
```

Comment:
Just a thought: why not putting this check into the `doc_hidden` function directly? Since you're doing this check every time, I think it'd make more sense.